lib/child_process.js

```diff
@@ -140,9 +140,21 @@ exports.exec = function(command /*, options, callback*/) {
                           opts.callback);
 };
 
-Object.defineProperty(exports.exec, customPromisifyArgs,
-                      { value: ['stdout', 'stderr'], enumerable: false });
-
+Object.defineProperty(exports.exec, util.promisify.custom, {
+  enumerable: false, value: function(...args) {
+    const promise = createPromise();
+    exports.exec.call(this, ...args, (err, stdout, stderr) => {
```

Comment:
I don’t think you need to use `.call`; `exec` should work no matter what `this` refers to